include/dgl/aten/macro.h

```diff
@@ -309,7 +309,9 @@
   });
 
 #define CHECK_VALID_CONTEXT(VAR1, VAR2)                          \
-  CHECK(((VAR1)->ctx == (VAR2)->ctx) || (VAR1).IsPinned())       \
+  CHECK(                                                         \
+      ((VAR1)->ctx == (VAR2)->ctx) || (VAR1).IsPinned() ||       \
+      ((VAR1).NumElements() == 0))                               \
```

Comment:
Do you think if it would be possible or better to check 0-element and return early [here](https://github.com/dmlc/dgl/blob/e28f07812bae274f5ba49d26629de5541c9cff60/src/array/array.cc#L120)?

Reply:
I agree. But I find we need to return early for many other functions, such as `dgl::aten::CSRSliceRows()`...

I will add a comment to this macro.